client/src/templates/Challenges/components/completion-modal.tsx

```diff
@@ -77,21 +77,32 @@ export function getCompletedPercent(
   currentBlockIds: string[] = [],
   currentChallengeId: string
 ): number {
-  completedChallengesIds = completedChallengesIds.includes(currentChallengeId)
-    ? completedChallengesIds
-    : [...completedChallengesIds, currentChallengeId];
-
-  const completedChallengesInBlock = completedChallengesIds.filter(id => {
-    return currentBlockIds.includes(id);
-  });
-
+  const completedChallengesInBlock = getCompletedChallengesInBlock(
+    completedChallengesIds,
+    currentBlockIds,
+    currentChallengeId
+  );
   const completedPercent = Math.round(
-    (completedChallengesInBlock.length / currentBlockIds.length) * 100
+    (completedChallengesInBlock / currentBlockIds.length) * 100
   );
 
   return completedPercent > 100 ? 100 : completedPercent;
 }
 
+function getCompletedChallengesInBlock(
+  completedChallengesIds: string[],
+  currentBlockChallengeIds: string[],
+  currentChallengeId: string
+) {
+  const currentChallengeAlreadyCompleted =
+    completedChallengesIds.includes(currentChallengeId);
+  return (
+    completedChallengesIds.filter(challengeId =>
+      currentBlockChallengeIds.includes(challengeId)
+    ).length + +(!currentChallengeAlreadyCompleted && 1)
+  );
```

Comment:
I have probably made things worse here, but my brain cannot come up with the simplified version of this.

Reply:
I did read your comment to make the code more readable for ease of maintenance.

But I did not touch it because I do not have complete knowledge of the code base.

And was not sure which places to ammends.

Reply:
Oliver has simplified this.